internal/serve/serve.go

```diff
@@ -109,7 +107,12 @@ func (opts *ServeOptions) SetupDependencies() error {
 	}
 
 	// Setup Multi-Tenant Database when enabled
+	tssDBConnectionPool := dbConnectionPool
 	if opts.EnableMultiTenantDB {
+		deprecatedConnectionPool := dbConnectionPool
+		defer deprecatedConnectionPool.Close()
```

Comment:
Is the reason for this change to close the connection ? Because I don't see this being used somewhere else.

Reply:
The following lines are opening a new connection pool, so I'm closing this one since it's not being used anywhere else.